app/vep/vep_resources.py

```diff
@@ -54,12 +61,30 @@ async def submit_vep(request: Request):
         else:
             raise Exception
     except MaxBodySizeException:
-        return HTTPException(
-            status_code=status.HTTP_413_REQUEST_ENTITY_TOO_LARGE,
-            detail="Maximum file size limit exceeded.",
-        )
+        return response_error_handler(result={"status": 413})
     except Exception as e:
-        return HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-            detail="Could not submit VEP job/workflow.",
-        )
+        return response_error_handler(result={"status": 500})
+
+
+@router.get("/submissions/{job_id}/status", name="submission_status")
+async def submit_vep(request: Request, job_id: str):
+    try:
+        workflow_status = await get_workflow_status(job_id)
+        return JSONResponse(content=workflow_status.dict())
+
+    except HTTPError as http_error:
+        if http_error.response.status_code == 403:
```

Comment:
Didn't we see that Seqera's response to a missing workflow id can also be a `400`? We would also want to transform that into a 404.

Reply:
Done!